src/v2/Components/SaleDetailTimer.tsx

```diff
@@ -0,0 +1,110 @@
+import { SaleDetailTimer_sale } from "v2/__generated__/SaleDetailTimer_sale.graphql"
+import { createFragmentContainer, graphql } from "react-relay"
+import * as React from "react"
+import { Flex, Text } from "@artsy/palette"
+import { useTimer } from "v2/Utils/Hooks/useTimer"
+import moment from "moment"
+export interface SaleDetailTimerProps {
+  sale: SaleDetailTimer_sale
+}
+
+function relativeDaysUntil(parsedDays, suffix) {
+  if (parsedDays === 0) {
+    return `${suffix} Today`
+  } else if (parsedDays === 1) {
+    return `${parsedDays} Day Until ${suffix}`
+  } else {
+    return `${parsedDays} Days Until ${suffix}`
+  }
+}
+
+export const SaleDetailTimer: React.FC<SaleDetailTimerProps> = ({ sale }) => {
+  const endAt = sale?.endAt
+
+  const startAt = sale?.startAt
+
+  const endedAt = sale?.endedAt
+
+  const { time } = useTimer(endAt!, startAt!)
+
+  if (!endAt) {
+    return null
+  }
```

Comment:
#minor but you can move up to line 23

Reply:
You can't actually because of the hook to useTimer, you'll get a linting error if you try and move it up ` React Hook "useTimer" is called conditionally.`